test/cors.test.js

```diff
@@ -160,9 +160,25 @@ test('Dynamic origin resolution (errored)', t => {
   }
   fastify.register(cors, { origin })
 
-  fastify.get('/', (req, reply) => {
-    reply.send('ok')
+  fastify.inject({
+    method: 'GET',
+    url: '/',
+    headers: { origin: 'example.com' }
+  }, (err, res) => {
+    t.error(err)
+    t.strictEqual(res.statusCode, 500)
   })
+})
+
+test('Dynamic origin resolution (invalid result)', t => {
+  t.plan(3)
+
+  const fastify = Fastify()
+  const origin = (header, cb) => {
+    t.strictEqual(header, 'example.com')
+    cb(null, undefined)
```

Comment:
@RafaelGSS this test covers falsy values. The test is in the context of an origin function that returns a falsy value, but covers that case. Is ok for you?